ui/pages/desktop-error/render-desktop-error.js

```diff
@@ -61,10 +74,57 @@ export function renderDesktopError({
     );
   };
 
+  const getEventNameById = (id) => {
+    let eventName;
+    switch (id) {
+      case DESKTOP_ERROR_BUTTON_DOWNLOAD_ID:
+        eventName = 'Download MetaMask Desktop';
+        break;
+      case DESKTOP_ERROR_BUTTON_OPEN_OR_DOWNLOAD_ID:
+        eventName = 'Open MetaMask Desktop';
+        break;
+      case DESKTOP_ERROR_BUTTON_DISABLE_ID:
+        eventName = 'Disable MetaMask Desktop';
+        break;
+      case DESKTOP_ERROR_BUTTON_UPDATE_ID:
+        eventName = 'Update MetaMask Desktop';
+        break;
+      case DESKTOP_ERROR_BUTTON_NAVIGATE_SETTINGS_ID:
+        eventName = 'Return to Settings Page';
+        break;
+      case DESKTOP_ERROR_BUTTON_UPDATE_EXTENSION_ID:
+        eventName = 'Update MetaMask Extension';
+        break;
+      case DESKTOP_ERROR_BUTTON_RESTART_ID:
+        eventName = 'Restart MetaMask';
+        break;
+      default:
+        eventName = 'Return MetaMask Home';
+        break;
+    }
+    return eventName;
+  };
+
   const renderCTA = (id, text, onClick) => {
     return (
       <Box marginTop={6}>
-        <Button type="primary" onClick={onClick ?? noop} id={id}>
+        <Button
+          type="primary"
+          onClick={() => {
+            if (onClick) {
+              onClick();
+              if (typeof trackEvent === 'function') {
+                trackEvent({
+                  category: EVENT.CATEGORIES.DESKTOP,
+                  event: `${getEventNameById(id)} Button Clicked`,
```

Comment:
using dynamic event names is frowned upon as it will create N new tables.  can this be a single event name and use dynamic id (if truely necessary) as a property instead? (though even that is less than ideal, better if you run this through a defined list of possible values)

Reply:
:pray:  Great thanks!
I changed it to a single event `Desktop Button Clicked` and also included an additional parameter in the `renderCTA` that will be the value of the property to identify the CTA performed by the user.